BlockChainRunner.java

```diff
@@ -0,0 +1,354 @@
+package com.karolismed.simple_blockchain.blockchain;
+
+import com.karolismed.simple_blockchain.blockchain.model.block.Block;
+import com.karolismed.simple_blockchain.blockchain.model.User;
+import com.karolismed.simple_blockchain.blockchain.model.transaction.Transaction;
+import com.karolismed.simple_blockchain.blockchain.model.transaction.TransactionInput;
+import com.karolismed.simple_blockchain.blockchain.model.transaction.TransactionOutput;
+import com.karolismed.simple_blockchain.blockchain.model.transaction.UnspentOutput;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.ThreadLocalRandom;
+import org.apache.commons.collections4.MultiValuedMap;
+import org.apache.commons.collections4.multimap.HashSetValuedHashMap;
+import org.apache.commons.lang3.tuple.Pair;
+
+import static com.karolismed.simple_blockchain.constants.GlobalConstants.BLOCK_ZERO_HASH;
+import static com.karolismed.simple_blockchain.constants.GlobalConstants.GENESIS_BLOCK_DIFFICULTY;
+import static com.karolismed.simple_blockchain.constants.GlobalConstants.REGULAR_BLOCK_DIFFICULTY;
+import static com.karolismed.simple_blockchain.constants.GlobalConstants.TRANSACTION_POOL_SIZE;
+import static com.karolismed.simple_blockchain.utils.StringHelper.log;
+import static java.util.Objects.isNull;
+
+public class BlockChainRunner {
+
+    private final BlockChainSetupService setupService;
+    private final MerkleTreeConstructor merkleTreeConstructor;
```

Comment:
Should avoid such long variable names, as they seem to be excessive and filling up the code itself.

An example to change the naming would be _merkleTrConstr_

Reply:
The most important thing about a variable name is that it's descriptive and easily readable. Here I'd argue `merkleTreeConstructor` is more readable than `merkleTrConstr`. More info on why abbreviations should be avoided in variable names: https://blog.boot.dev/clean-code/naming-variables/#8-dont-use-abbreviations-or-acronyms-without-sufficient-context